src/data/matrix/notation.lean

```diff
@@ -41,90 +33,17 @@ Examples of usage can be found in the `test/matrix.lean` file.
 namespace matrix
 
 universe u
-variables {α : Type u}
+variables {α : Type u} {o n m : ℕ} {m' n' o' : Type*}
 
 open_locale matrix
 
-section matrix_notation
-
-/-- `![]` is the vector with no entries. -/
-def vec_empty : fin 0 → α :=
-fin_zero_elim
-
-/-- `vec_cons h t` prepends an entry `h` to a vector `t`.
-
-The inverse functions are `vec_head` and `vec_tail`.
-The notation `![a, b, ...]` expands to `vec_cons a (vec_cons b ...)`.
--/
-def vec_cons {n : ℕ} (h : α) (t : fin n → α) : fin n.succ → α :=
-fin.cons h t
-
-notation `![` l:(foldr `, ` (h t, vec_cons h t) vec_empty `]`) := l
-
-/-- `vec_head v` gives the first entry of the vector `v` -/
-def vec_head {n : ℕ} (v : fin n.succ → α) : α :=
-v 0
-
-/-- `vec_tail v` gives a vector consisting of all entries of `v` except the first -/
-def vec_tail {n : ℕ} (v : fin n.succ → α) : fin n → α :=
-v ∘ fin.succ
-
-variables {m n : ℕ}
-
-/-- Use `![...]` notation for displaying a vector `fin n → α`, for example:
-
-```
-#eval ![1, 2] + ![3, 4] -- ![4, 6]
-```
--/
-instance [has_repr α] : has_repr (fin n → α) :=
-{ repr := λ f, "![" ++ (string.intercalate ", " ((list.fin_range n).map (λ n, repr (f n)))) ++ "]" }
-
 /-- Use `![...]` notation for displaying a `fin`-indexed matrix, for example:
 
 ```
 #eval ![![1, 2], ![3, 4]] + ![![3, 4], ![5, 6]] -- ![![4, 6], ![8, 10]]
```

Comment:
This example is weird, because lean doesn't even know this has type `matrix` so doesn't use this instance. Out of scope I guess though, since it was here before.